app/experimenter/nimbus-ui/src/hooks/useCommonForm.tsx

```diff
@@ -0,0 +1,143 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from "react";
+import { RegisterOptions, FieldError } from "react-hook-form";
+import Form from "react-bootstrap/Form";
+import { useForm } from "react-hook-form";
+
+// exported for testing
+export const overviewFieldNames = [
+  "name",
+  "hypothesis",
+  "application",
+  "publicDescription",
+] as const;
+
+export const audienceFieldNames = [
+  "channel",
+  "firefoxMinVersion",
+  "targetingConfigSlug",
+  "populationPercent",
+  "totalEnrolledClients",
+  "proposedEnrollment",
+  "proposedDuration",
+] as const;
+
+export const metricsFieldNames = [
+  "primaryProbeSetIds",
+  "secondaryProbeSetIds",
+] as const;
+
+type OverviewFieldNames = typeof overviewFieldNames[number];
+type AudienceFieldNames = typeof audienceFieldNames[number];
+type MetricsFieldNames = typeof metricsFieldNames[number];
+
+type FieldNames = OverviewFieldNames | AudienceFieldNames | MetricsFieldNames;
```

Comment:
Seems like these should live with their respective forms? And FieldNames as a parameterized type at the point of using `useCommonForm`? (e.g. `useCommonForm<OverviewFieldNames>` or suchlike?)

Reply:
Ooh that seems reasonable and makes it less cumbersome to update should we ever add more forms

Reply:
@lmorchard Hmm yeah, I can move them/look into that. (I was waiting for you to take a look at this PR anyway so feel free to poke more 🙂)

Reply:
Also, I was curious whether my suggestion was actually harder than it seemed, so I gave it a go as an exercise. I think it ended up being pretty simple. If you haven't already tried it, this seems not to break things:

https://github.com/mozilla/experimenter/compare/issues/4021...lmorchard:issues/4021?expand=1